pkg/dev/interface.go

```diff
@@ -17,44 +16,29 @@ type Client interface {
 	// If buildCommand is set, this will look up the specified build command in the Devfile. Otherwise, it uses the default one.
 	// If runCommand is set, this will look up the specified run command in the Devfile and execute it. Otherwise, it uses the default one.
 	// Returns the status of the started component
-	Start(
-		devfileObj parser.DevfileObj,
-		componentName string,
-		namespace string,
-		ignorePaths []string,
-		path string,
-		debug bool,
-		buildCommand string,
-		runCommand string,
-		randomPorts bool,
-		errOut io.Writer,
-		fs filesystem.Filesystem,
-	) (watch.ComponentStatus, error)
-
-	// Watch watches for any changes to the files under path while ignoring the files/directories in ignorePaths.
+	//
+	// It then watches for any changes to the files under path while ignoring the files/directories in ignorePaths.
 	// It logs messages to out and uses the Handler h to perform push operation when anything changes in path.
 	// It uses devfileObj to notify user to restart odo dev if they change endpoint information in the devfile.
 	// If debug is true, the debug command will be started after a sync, or the run command by default.
 	// If buildCommand is set, this will look up the specified build command in the Devfile. Otherwise, it uses the default one.
 	// If runCommand is set, this will look up the specified run command in the Devfile and execute it. Otherwise, it uses the default one.
 	// componentStatus is the status returned from the call to the Start Method
-	Watch(
-		devfilePath string,
+	Start(
+		ctx context.Context,
 		devfileObj parser.DevfileObj,
 		componentName string,
 		path string,
+		devfilePath string,
 		ignorePaths []string,
-		out io.Writer,
-		h Handler,
-		ctx context.Context,
 		debug bool,
 		buildCommand string,
 		runCommand string,
-		variables map[string]string,
 		randomPorts bool,
 		watchFiles bool,
+		variables map[string]string,
+		out io.Writer,
 		errOut io.Writer,
-		componentStatus watch.ComponentStatus,
 	) error
```

Comment:
IMO, we should separate optional parameters into a struct. This is too large a number of arguments for the function, no?